### DISCUSSION TITLE
Composition of sprites on an entity

### DISCUSSION THREAD

**Discussion by gmoller** [bevyengine/bevy#2870] [Q&A] [answered]
Hi,

Could someone please point me in the right direction for creating an entity with a composition of sprites instead of just one ColorMaterial? I'm wondering if it would be possible to change the composition of the sprites dynamically  based on the state of an entity, such as when it is selected or hovered over, etc.

Thanks,
Greg

**Comment by payload** [comment, marked as answer]
Make each sprite its own entity so each can have their ColorMaterial and so on. But have a root entity holding everything together by making each sprite a child of the parent entity.

Later you can access each child sprite with a query when you know the entity. I sometimes use marker components for each child or like in the example below a component holding the Entity handles to each sprite in the composition.

Example:
```
let left_arm = commands.spawn_bundle(...).id();
let right_arm = commands.spawn_bundle(...).id();
let head = commands.spawn_bundle(...).id();

// root entity and component holding everything together
// struct Body { left_arm: Entity, right_arm: Entity, head: Entity };
commands.spawn_bundle((Transform::identity(), GlobalTransform::identity(), Body { left_arm, right_arm, head })).push_children(&[left_arm, right_arm, head]);

// later in a system you can access and modify each ColorMaterial for example with
color_material_query.get_mut(body.left_arm)
```

**Reply by gmoller** [reply]
Thank you much!

**Reply by gmoller** [reply]
Although I couldn't get it to compile as:

commands.spawn_bundle((Transform::identity(), GlobalTransform::identity(), Body { left_arm, right_arm, head }).push_children(&[left_arm, right_arm, head]);

It seems to compile with:
commands.spawn_bundle((Transform::identity(), GlobalTransform::identity(), Body { left_arm, right_arm, head })).push_children(&[left_arm, right_arm, head]);

**Reply by payload** [reply]
yes, missing closing parenthesis. updated comment
